include/nuttx/sensors/sensor.h

```diff
@@ -827,6 +827,23 @@ extern "C"
 #define EXTERN extern
 #endif
 
+/****************************************************************************
+ * Name: sensor_remap_vector_raw16
+ *
+ * Description:
+ *   This function remap the sensor data according to the place position on
+ *   board. The value of place is determined base on g_remap_tbl.
+ *
+ * Input Parameters:
+ *   in    - A pointer to input data need remap.
+ *   out   - A pointer to output data.
+ *   place - The place position of sensor on board.
+ *
+ ****************************************************************************/
+
+void sensor_remap_vector_raw16(FAR const int16_t *in, FAR int16_t *out,
```

Comment:
IMO that should be a part of `sensor_ioctl` call.

Reply:
Generally, the position of sensor is fixed on board, so it's best to configure the direction when initialization and ensure that the initial data is correct.

Reply:
Then maybe be it can be a part of `struct sensor_lowerhalf_s`? I do not understand this API and the usage. You added some global array and perform some "remapping", but I do not understand what should be remapped. I mean why this function should be placed in `drivers/sensors/sensor.c` and not in your board source file?

Reply:
I hope this function as a common function to remap between sensor vector data and position on board.
The g_remap_tbl, this array contains all positions possibly except sensor is placed  in vertical direction on the board.

```
struct sensor_axis_map_s
{
  int8_t src_x;
  int8_t src_y;
  int8_t src_z;

  int8_t sign_x;
  int8_t sign_y;
  int8_t sign_z;
};
```

src_x, src_y, src_z is used to represent the axis data swaps, 
sign_x, sign_y, sign_z is used  to represent the axis data inverse.

you can using the function follow this examples:

```
#define LSM6DSO_VECTOR_REMAP      2

union 
{
  int16_t i16bit[3];    
  uint8_t u8bit[6];     
}  temp;

lsm6dso_spi_read(priv, regaddr, temp.u8bit, 6);
sensor_remap_vector_raw16(temp.i16bit, temp.i16bit, LSM6DSO_VECTOR_REMAP);      

value->x = temp.i16bit[0] *priv->dev[LSM6DSO_XL_IDX].factor;
value->y = temp.i16bit[1] *priv->dev[LSM6DSO_XL_IDX].factor;
value->z = temp.i16bit[2] *priv->dev[LSM6DSO_XL_IDX].factor;
```

Reply:
Sorry, but I still do not understand why this API should be placed in `include/nuttx/sensors/sensor.h`. The header file contains generic sensor driver description, but this API seems to be sensor type specific. Or maybe I'm missing something? Let's talk with a use cases. For example I have `SENSOR_TYPE_AMBIENT_TEMPERATURE` sensor installed on the board. How this API should be applied in this case?

Reply:
This function is used to vector sensor, examples:accel, gyro and magnetic, the temp sensor don't need to use it to remap data. If the function is exist for every vector sensor driver, the code can't reuse.

Reply:
This change confuse me a lot. I'm feeling like I'm missing some design / use case parts to understand it. I would like to have an example from one of existing sensor drivers that use this API. I failed to find neither `lsm6dso_spi_read` nor `LSM6DSO_XL_IDX`, so the picture is still blurred for me.

Reply:
Okay, i close this pr first, and submit it along with driver code.

Reply:
Some additional feedback from me:
```
static const struct sensor_axis_map_s g_remap_tbl[] =
{
  { 0, 1, 2,  1,  1,  1 }, /* P0 */
  { 1, 0, 2,  1, -1,  1 }, /* P1 */
  { 0, 1, 2, -1, -1,  1 }, /* P2 */
  { 1, 0, 2, -1,  1,  1 }, /* P3 */
  { 0, 1, 2, -1,  1, -1 }, /* P4 */
  { 1, 0, 2, -1, -1, -1 }, /* P5 */
  { 0, 1, 2,  1, -1, -1 }, /* P6 */
  { 1, 0, 2,  1,  1, -1 }, /* P7 */
};
...
remap = &g_remap_tbl[place];
```
where `place - The place position of sensor on board.`. The "place position" is not obvious. Is it "top", "botton", "left side", etc. or anything else? I mean that for you as an author it might be obvious, but me as a reviewer have to guess how P0 differs from P6 (i mean not in `src` and `sign` point of veiw).

Reply:
@pkarashchenko I add some description for these body corrdinate:
```
/* Body coordinate system position P0:
 *
 *          +y
 *          |
 *          |
 *          |
 *          |
 *          .------>+x
 *         /
 *        /
 *       /
 *      /
 *     +z
 *
 */

#define SENSOR_BODY_COORDINATE_P0                   0

/* Body coordinate system position P1:
 *
 *          .------>+y
 *         /|
 *        / |
 *       /  |
 *      /   |
 *     +z  -x
 *
 */

#define SENSOR_BODY_COORDINATE_P1                   1

/* Body coordinate system position P2:
 *
 * -x<------.
 *         /|
 *        / |
 *       /  |
 *      /   |
 *     +z  -y
 *
 */

#define SENSOR_BODY_COORDINATE_P2                   2

/* Body coordinate system position P3:
 *
 *          +x
 *          |
 *          |
 *          |
 *          |
 * -y<------.
 *         /
 *        /
 *       /
 *      /
 *     +z
 *
 */

#define SENSOR_BODY_COORDINATE_P3                   3

/* Body coordinate system position P4:
 *
 *          +y  -z
 *          |   /
 *          |  /
 *          | /
 *          |/
 * -x<------.
 *
 */

#define SENSOR_BODY_COORDINATE_P4                   4

/* Body coordinate system position P5:
 *
 *          +y  -z
 *          |   /
 *          |  /
 *          | /
 *          |/
 * -x<------.
 *
 */

#define SENSOR_BODY_COORDINATE_P5                   5

/* Body coordinate system position P6:
 *
 *              -z
 *              /
 *             /
 *            /
 *           /
 * -x<------.
 *          |
 *          |
 *          |
 *          |
 *         -y
 *
 */

#define SENSOR_BODY_COORDINATE_P6                   6

/* Body coordinate system position P6:
 *
 *         +x    -z
 *          |    /
 *          |   /
 *          |  /
 *          | /
 *          .------->y
 *
 */

#define SENSOR_BODY_COORDINATE_P7                   7
```